chain/chain/src/flat_storage_resharder.rs

```diff
@@ -640,6 +640,19 @@ impl FlatStorageResharder {
                 batch_id = ?num_batches_done)
             .entered();
             let chain_final_head = chain_store.final_head()?;
+
```

Comment:
mini nit: It's a bit weird to have almost the same condition duplicated in the inner and outer loop. Maybe consider refactoring it to only have one loop but instead to commit the changes once some counter reaches the `catch_up_blocks` threshold?

Reply:
Yes it is a bit unfortunate to have two conditions, I'll at least merge them together.
I think I'll push a bigger refactor into another PR, if I can make it happen 👍